subprojects/platform-native/src/docs/guide-templates/building-native-application/README.adoc

```diff
@@ -0,0 +1,172 @@
+= Building {C Applications}
+:summary: Building {C application}s.
+:type: guide-chapter
+:tags: guide, {application}, sources, native, {C}, gradle
+:category: {C}
+:description: Learn how to build {an application} implemented in {C} using the Gradle Nokee plugins.
+
+This guide demonstrates how to create a {C application} using the Nokee plugins for Gradle.
+You can follow the guide step-by-step to create a new project from scratch or download the complete project using the links above.
+
+== What we'll build
+
+We’ll generate a {C application} that follows Nokee and Gradle’s conventions.
+
+== What we’ll need
+
+* A text editor or IDE - for example https://www.jetbrains.com/idea/download/[IntelliJ IDEA]
+* A Java Development Kit (JDK), version 8 or higher - for example https://adoptopenjdk.net/[AdoptOpenJDK]
+* The latest https://gradle.org/install[Gradle distribution]
+* An installed C compiler.
+ See which link:TODO[C toolchains] are supported by Nokee.
+
+== 1. Create a project folder
+
+First, let's create an empty folder for our project and use the build-in `wrapper` task to create a Gradle wrapper script, e.g. `gradlew`.
+
+[source,terminal]
+----
+$ mkdir demo
+$ cd demo
+$ gradle wrapper
+
+BUILD SUCCESSFUL
+1 actionable tasks: 1 executed
+$ tree .
+├── gradle       // <1>
+│   └── wrapper
+│       ├── gradle-wrapper.jar
+│       └── gradle-wrapper.properties
+├── gradlew      // <2>
+└── gradlew.bat  // <2>
+
+2 directories, 4 files
+$ touch settings.gradle // <3>
+
+----
+
+<1> Generated folder for wrapper files
+
+<2> Gradle wrapper start scripts
+
+<3> Create an empty Gradle workspace buildscript
+
+We now have a Gradle project setup.
+Let's configure {C application} capability to the project.
+
+== 2. Configure {C application} capability
+
+First, create the project buildscript, e.g. `build.gradle`.
+
+build.gradle
+
+[source,groovy]
+----
+plugins {  // <1>
+	id 'dev.nokee.c-application' // <2>
+}
+----
+
+<1> We use the plugin block to apply plugins that extends the project capability.
+We can apply any Gradle project plugin here.
+Learn more about the link:TODO[plugins block in Gradle's documentation].
+Refer to each plugin's reference chapter to learn more about each individual plugins.
+
+<2> We apply Nokee's {C application} plugin.
+Learn more about link:TODO[this plugin at its reference chapter].
+
+NOTE: Gradle support two domain specific language (DSL): Groovy DSL and Kotlin DSL.
+For this guide, we will use Groovy DSL which is closer in syntax to {C} language.
+Learn more about Gradle DSL language in the link:TODO[Getting started chapter].
+
+== 3. Add {C} code
+
+By default, Nokee uses link:TODO[Gradle's conventional source layout].
+All {C} application code is located at `src/main/c` and its private headers are located at `src/main/headers`.
```

Comment:
🔲 The conventional source layout page is missing. The page would go on explaining what is the _conventional source layout_ with its reasoning. I tried to summarize the _layout_ as a straightforward description.

❌  We should call out to working with sources manual chapter (to understand the concept behind source layout) and the sample that shows how to configure the source layout. It's important to mention to avoid users thinking the layout is not configurable.